java/src/main/java/org/whispersystems/signalservice/loki/api/opengroups/PublicChatAPI.kt

```diff
@@ -287,6 +291,42 @@ class PublicChatAPI(userPublicKey: String, private val userPrivateKey: ByteArray
         }
     }
 
+    public fun updateOpenGroupProfileIfNeeded(channel: Long, server: String, groupId: String, info: LokiPublicChatInfo, groupDatabase: LokiGroupDatabaseProtocol, forceUpdate: Boolean) {
+        //Save user count
+        apiDatabase.setUserCount(channel, server, info.memberCount)
+
+        //Update display name
+        groupDatabase.updateTitle(groupId, info.displayName)
+
+        //Download and update profile picture if needed
+        val oldAvatarURL = apiDatabase.getOpenGroupAvatarURL(channel, server)
+        if (forceUpdate || !Objects.equals(oldAvatarURL, info.profilePictureURL)) {
```

Comment:
@nielsandriesse ah right, thanks for bringing it up! I keep thinking in Java way, where `X.equals(Y)` may result in NPE in case `X` is null. Seems like Kotlin's `!=`operator is safe in that regard.